packages/salesforcedx-vscode-lwc/package.json

```diff
@@ -81,6 +81,7 @@
     "test:vscode-insiders-integration": "cross-env CODE_VERSION=insiders npm run test:vscode-integration"
   },
   "activationEvents": [
+    "workspaceContains:modules",
```

Comment:
@lcampos this is needed to activate the extension in the subfolders as we discussed (not only for tests but general usage of LWC extension) - feel free to suggest any alternative

Reply:
This new event would still cause the activation of the extensions in projects that are not salesforce or core projects, causing a regression to these issues we've fixed (#1988 and #2065). 
I'd have to take a look at how we structure individual lwc modules in core and see if there's something we can leverage here. @mysticflute do you have any suggestions ?

Reply:
@lcampos IMHO the regression risk is relatively low here since it was checking the _top-level_ folder `module`, which might not be popular among non-salesforce projects. VS Code doesn't seem to provide an [activation event](https://code.visualstudio.com/api/references/activation-events) to check outside of workspace folder either. Maybe it's possible to execute a [command](https://code.visualstudio.com/api/references/activation-events#onCommand) for activation, based on the parent folder check, but I'm not sure it's worth the effort. If we want to be very safe here, there need to be a null check somewhere regarding #2065.

Reply:
Regarding #1988, looks like this issue still presents, as of 48.11.0. Briefly took a look and I think it might be because that `salesforcedx-vscode-lightning` contributes to `html` language. `activate` hasn't been called. Only when I disable "Aura Components" extension file syntax highlighting works for embedded html in PHP.

In my opinion there isn't much we can do here, except recommending users to disable "Aura Components" extension within the workspace.

Reply:
regarding core workspaces, I guess it depends on the available options for `activationEvents`. Core modules should have a pom.xml file that extends from the salesforce pom. If possible that could be something we could check for, but it doesn't seem we can do a check like that here.

If we can only check for the presence of files, if it's possible to check the root folder then we can check that it contains the `modules` folder, `pom.xml` file, and `build` directory which should is pretty specific. In this case the better long term IMO if we add a configuration file that should be present and we check for that, e.g., `salesforce-vscode.config.json`. And make this file part of the template for component modules or have blt automatically add it.

That is more work though. While we have talked about possible needing a configuration file I don't think we need it yet. So what I'm leaning towards is letting the extension activate, and in the extension activation code we should have more checks to see if it should actually do anything there.

Reply:
@mysticflute Good point. I believe the extension is already checking if the workspace [isLWC](https://github.com/forcedotcom/lightning-language-server/blob/0e7dbe94671a62991109cadc95f22aa0d948a0ae/packages/lightning-lsp-common/src/shared.ts#L35) in [activation code](https://github.com/forcedotcom/salesforcedx-vscode/blob/d88eb17761e96bbfdf27aa99ccfcdd2798bf34f2/packages/salesforcedx-vscode-lwc/src/index.ts#L72-L83), if activationMode config is set to `autodetect` (which is the default config). If activationMode config is set to `always` it always activates.

Reply:
Moving the activation part to a new work item tracking it W-7509875, as discussed offline with @lcampos